tools/locale-nav/locale-selector.css

```diff
@@ -0,0 +1,139 @@
+:host {
+  font-family: 'Adobe Clean', adobe-clean, sans-serif;
+
+  --local-padding: 0 6px;
+}
+
+.locale-selector {
+  width: 100%;
+  overflow: auto;
+  font-size: 0.875rem;
+}
+
+.locale-header {
+  background-color: #f8f8f8;
+  display: flex;
+  justify-content: space-between;
+  padding: var(--local-padding);
+  line-height: 36px;
+  text-transform: uppercase;
+  font-weight: 700;
+}
+
+.actions {
+  display: grid;
+  grid-template-columns: 48px 60px 48px;
+  text-align: center;
+}
+
+.action {
+  height: 42px;
+  width: 42px;
+  margin: 2px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+}
+
+.detail {
+  display: flex;
+  justify-content: space-between;
+  padding: var(--local-padding);
+  line-height: 46px;
+  font-size: 18px;
+  text-transform: uppercase;
+}
+
+li::after {
+  content: "";
+  display: block;
+  height: 1px;
+  background: #d1d1d1;
+  width: 100%;
+  flex: 0 0 auto;
+}
+
+.action .details {
+  border: 0;
+  clip: rect(1px, 1px, 1px, 1px);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  width: 1px; 
+}
+
+.action .icon {
+  height: 26px;
+  width: 26px;
+}
+
+.action .icon-html {
+  background: url('/tools/locale-nav/img/Smock_FileHTML_18_N.svg') center / 26px no-repeat;
+}
+
+.action .icon-aem {
+  border-radius: 5px;
+  background-color: #959595;
+  background-image: url('/tools/locale-nav/img/Smock_ExperienceCloud_24_N.svg');
+}
+
+.action .icon-aem.status-200 {
+  background-color: #fa0f00;
+}
+
+.action .icon-aem.status-404 {
+  pointer-events: none;
+  background-color: #fa0f00;
+  opacity: 0.25;
+}
+
+ul.locales {
+  margin: 0;
+  padding: 0;
+  list-style: none;
+}
+
+.locale-search-wrapper {
+  position: relative;
+}
+
+.locale-search {
+  width: 100%;
+  background-color: #f8f8f8;
+  display: block;
+  font-family: var(--body-font-family);
+  border: none;
+  margin: 0;
+  padding: var(--local-padding);
+  line-height: 36px;
+  text-transform: uppercase;
+  font-weight: 700;
+  box-sizing: border-box;
+  transition: padding .2s ease-in-out;
+}
+
+.locale-search-icon {
+  position: absolute;
+  content: '';
+  width: 36px;
+  height: 36px;
+  left: -36px;
+  top: 0;
+  background: url('/tools/locale-nav/img/spectrum-search.svg') center / 24px no-repeat;
+  transition: left .2s ease-in-out;
+}
+
+.locale-search:hover + .locale-search-icon, .locale-search:focus + .locale-search-icon {
+  left: 6px;
+}
+
+.locale-search:hover, .locale-search:focus {
+  padding: 0 42px;
+}
+
+.locale-search::placeholder {
+  color: #676767;
+}
```

Comment:
Some generic names in here like `action` `detail` `li:after` do you need to namespace any of these to avoid cascade issues?

Reply:
This is using the shadow DOM with adoptedStyleSheets, it hasn't been an issue but I could see namespacing it.